sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -1957,6 +2125,84 @@ public Duration getAllowedTimestampSkew() {
 
       pipeline.run();
     }
+
+    @Test
+    @Category({ValidatesRunner.class, UsesTimersInParDo.class})
+    public void testProcessElementSkew() {
+      List<KV<String, Duration>> durations =
+          Arrays.asList(KV.of("0", Duration.millis(0L)), KV.of("2", Duration.millis(1L)));
+      PCollection<KV<String, Duration>> input =
+          pipeline.apply("create", Create.timestamped(durations, Arrays.asList(0L, 1L)));
+
+      PCollection<String> noSkew =
+          input.apply("noSkew", ParDo.of(new ProcessElementTimestampSkewingDoFn(Duration.ZERO)));
+      PAssert.that(noSkew)
```

Comment:
The logic within `testOnTimerTimestampSkew` is much simpler to understand because the skew and noskew variants are differentiated.

Do you mind splitting this one up similarly?